module/VuFind/src/VuFind/Content/Covers/LocalFile.php

```diff
@@ -0,0 +1,150 @@
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\Content\Covers;
+
+/**
+ * Local file cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Leila Gonzales <lmg@agiweb.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class LocalFile extends \VuFind\Content\AbstractCover
+{
+
+    /**
+     * Base filepath for LocalFile
+     *
+     * @var string
+     */
+    protected $basePath;
+
+    /**
+     * Filename for LocalFile
+     *
+     * @var string
+     */
+    protected $fileName;
+
+    /**
+     * Image type for LocalFile
+     *
+     * @var string
+     */
+    protected $imageType;
+
+    /**
+     * Does this plugin support the provided ID array?
+     *
+     * @param array $ids IDs that will later be sent to load() -- see below.
+     *
+     * @return bool
+     */
+    public function supports($ids)
+    {
+        return true;
+    }
+
+    /**
+     * VuFind configuration
+     *
+     * @var Config
+     */
+    protected $config;
+
+    /**
+     * Constructor
+     *
+     * @param Config $config VuFind configuration
+     */
+    public function __construct($config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * Get image location from local file storage.
+     *
+     * @param string $key  local file directory path
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        //Get settings from config.ini file
+        $config = $this->config;
+        $basePath = isset($config->$key->filepath)
+            ? $config->$key->filepath : null;
+        $fileNameToken = isset($config->$key->filename)
+            ? $config->$key->filename : null;
+        $imageType = isset($config->LocalFile->imagetype)
+            ? $config->$key->imagetype : null;
+
+        if (!isset($basePath) || !isset($fileNameToken) || !isset($imageType)) {
+            return false;
+        }
+
+        // Check base path for $VUFIND env variables
+        $vufindHome =  getenv('VUFIND_HOME');
+        $vufindLocal =  getenv('VUFIND_LOCAL_DIR') 
+        $basePath = str_replace("\$VUFIND_HOME", $vufindHome, $basePath);
+        $basePath = str_replace("\$VUFIND_LOCAL_DIR", $vufindLocal, $basePath);
+
+        // Add slash to basePath if it doesn’t exist
+        if (substr($basePath, -1) != "/") {
+            $basePath = $basePath."/";
+        }
+
+        // match fileName to information in $ids array
+        if (isset($ids[$fileNameToken])) {
+            $fileName = $ids[$fileNameToken];
```

Comment:
If you want to go with my token idea, you could probably get away with something like this:

<pre>
// always use size token
$tokens = ['%size%'];
$replacements = [$size];
// convert other values to tokens:
foreach ($ids as $key => $val) {
    $tokens[] = '%' . $key . '%';
    $replacements[] = $val;
}
$fileName = str_replace($tokens, $replacements, $fileNameToken);
</pre>